tests/models/mistral/test_modeling_mistral.py

```diff
@@ -628,6 +628,100 @@ def test_speculative_generation(self):
         backend_empty_cache(torch_device)
         gc.collect()
 
+    @slow
+    def test_compile_static_cache(self):
+        # `torch==2.2` will throw an error on this test (as in other compilation tests), but torch==2.1.2 and torch>2.2
+        # work as intended. See https://github.com/pytorch/pytorch/issues/121943
+        if version.parse(torch.__version__) < version.parse("2.3.0"):
+            self.skipTest("This test requires torch >= 2.3 to run.")
+
+        NUM_TOKENS_TO_GENERATE = 40
+        EXPECTED_TEXT_COMPLETION = {
```

Comment:
if these are defined as self.xxxx then the test can be entirely copied from! 😉

Reply:
this means we can actually store this test in test_modelling_commong, only do it for models that have `_supports_static_cache`, and it checkes the expected completion with a XXXIntegrationModelTester.EXPECTED_TEXT_COMPLETION. This would allow de-duplicating, but is unrelated to the PR! cc @ydshieh might be nice?

Reply:
Sound good and doable!

Reply:
But we can do that clean up after this and my gemma PR.

Reply:
> this means we can actually store this test in test_modelling_commong, only do it for models that have `_supports_static_cache`, and it checkes the expected completion with a XXXIntegrationModelTester.EXPECTED_TEXT_COMPLETION. This would allow de-duplicating, but is unrelated to the PR! cc @ydshieh might be nice?

yeah, I think this can be another PR after